src/docs/spec-formats-tsv.md

```diff
@@ -0,0 +1,274 @@
+# The SSSOM/TSV serialisation format
+
+The SSSOM/TSV format is intended as the main format for exchanging SSSOM mapping set objects.
+
+The RECOMMENDED filename extension for a SSSOM/TSV file is `.sssom.tsv`, but SSSOM/TSV parsers MUST accept SSSOM/TSV files regardless of their extension.
+
+
+## Structure
+
+A SSSOM/TSV file contains one, and only one, mapping set object. It is made of two different parts:
+
+* the _metadata block_, which contains essentially all the slots of the [`MappingSet` class](MappingSet.md) except the `mappings` slot;
+* the _mappings block_ (also called the _TSV section_), which contains the individual mappings.
+
+A SSSOM/TSV file MUST NOT contain anything else than those two blocks.
+
+
+### Metadata block
+
+The metadata block is written as the [YAML 1.2](https://yaml.org/spec/1.2.2/) serialisation of the `MappingSet` object, except that the `mappings` slot is _not_ included (since it contains the mappings, that are serialised in the mappings block instead).
+
+The metadata block MUST appear at the beginning of the file. Every line of the block MUST be preceded by a `#` character; the `#` character MAY be followed by one or several space characters (U+0020) before the YAML content – if so, every line MUST have the same number of space characters.
+
+The metadata block ends with the first line that does not begin with a `#` character, which marks the beginning of the mappings block.
+
+The metadata block SHOULD only contains the slots that do have a value. SSSOM/TSV writers SHOULD skip slots with no value when serialising the mapping set object.
+
+#### Multi-valued slots with a single value
+
+As an exception to the standard YAML rules regarding the serialisation of sequences, a multi-valued slot that happens to contain a single value MAY be serialised as a scalar value rather than as sequence containing only one item.
+
+For example, a `creator_id` slot with the single value `ORCID:1111-2222-3333-4444` MAY be serialised as
+
+```yaml
+creator_id: "ORCID:1111-2222-3333-4444"
+```
+
```

Comment:
Given that we are using YAML, we are bound to face _a lot_ of variability anyway.

In fact that’s one of the main reasons for defining a “canonical” variant -- to reduce the variability.

But people are going to expect (are already expecting) to be able to do in the metadata block whatever YAML allows them to do, and I believe we should not stop them from doing so. Otherwise what would be the point of using YAML? We might as well define our own format to store the metadata block, without any of the variability of YAML.

YAML variability shouldn't be a problem for most implementations since they are most likely going to delegate parsing the metadata block to a full-blown YAML parser anyway (that's what both SSSOM-Py and SSSOM-Java are doing). In fact, trying to allow only a subset of what YAML allows is likely going to be _more difficult_ for implementations, because then they would have to override their YAML parser. I see no reason for doing that.

In the case at hand, the problem is the opposite: we are allowing something that YAML does not (representing a one-item list as a scalar value). But that form is widespread among SSSOM mapping sets. In fact, until [I corrected it a few months ago](https://github.com/mapping-commons/sssom/pull/362), this form was even found in the examples given in the spec! SSSOM-Py has always accepted that form when reading, and is still generating it when writing. I don't like that (it forced to do a lot of special-casing in SSSOM-Java’s parser), but we cannot outlaw that form now without breaking a lot of existing sets.

Reply:
Well, you are arguing for a less draconian spec so I wont stand in the way! I am fine with this, if need be, we can tighten the screws in a future version 2.0.

Reply:
One bit of variability that we could consider forbid, though, is the use of “advanced” YAML features such as tags, anchors and references. They may be nice, but I think they would be antithetical to SSSOM being a _Simple_ standard.

That is, how about explicitly forbidding things like

* Type tags

```yaml
comment: !!str This is explicitly typed as a string
```

Explicit typing should never be needed anyway since we have the LinkML schema to always tell us what type to expect for any slot.

* Non-type tags

```yaml
comment: !mytag This is a tagged comment
```

Such tags are currently not supported by SSSOM-Py, and I don’t think we’d lose much in forbidding their use entirely.

* Anchors and references

```yaml
creator_id: &the_creators
  - ORCID:1111-2222-3333-4444
author_id: *the_creators
```

Seemingly supported by SSSOM-Py, currently not by SSSOM-Java (because not supported by the underlying YAML library). Again, I don’t think we’d lose much in explicitly forbidding this kind of stuff.

Reply:
ok, yes. i never even knew these existed! definetly exclude imo

Reply:
> i never even knew these existed!

Pretty sure you’re not the only one. :) The complexity of YAML is regularly underestimated.